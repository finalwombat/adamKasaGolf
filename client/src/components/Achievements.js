import React from 'react'
import styled from 'styled-components'
import tamworthWin from '../img/tamworthWin.jpg'

const AchievementDiv = styled.div`
                      width: 90vw;
                      height: 90vh;
                      display: flex;
                      color: white;


                      & img {
  
                      }
                      
                      & div {
                        width: 30%;
                        margin: 60px;
                        align-self: center;
                      }

                      & a {
                        text-decoration: none;
                        background: #4286f4;
                        padding: 3px 8px 3px 8px;
                        color: black;
                        font-size: 1.3em;
                        transition: .5s ease;
                      }

                      & a:hover {
                        background: #2763c4;
                      }
`


const Achievements = (props) => {
  const Achievements = props.achievements.map( achievement => {
    return (
      <AchievementDiv>
        <div>
          <h2>{achievement.date}</h2>
          <h1>{achievement.title}</h1>
          <a href={achievement.link}>View</a>
        </div>
        <img src={tamworthWin} style={{height: '80%', width: '40%'}}/>
      </AchievementDiv>
    )
  })
  return (
      <div className="achievements">
        {Achievements}
      </div>
  )
}

export default Achievements