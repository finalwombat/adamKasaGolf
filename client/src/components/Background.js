import React from 'react'


const imgStyle = {
  height: '140vh',
  width: '100vw'
}

const Background = (props) => {
  return (
    <div className="backgroundComponent">
      <img src={props.img} style={imgStyle} alt="background" />
    </div>
  )
}

export default Background
