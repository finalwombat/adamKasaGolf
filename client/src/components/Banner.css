.banner {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 250px;
  height: 30px;
  left: 50px;
  position: relative;
  background: rgba(183, 183, 183, 1);
}
.banner:after {
  content: "";
  position: absolute;
  left: -15px;
  bottom: 0;
  width: 0;
  height: 0;
  border-left: 15px solid transparent;
  border-top: 15px solid rgba(183, 183, 183, 1);
  border-bottom: 15px solid rgba(183, 183, 183, 1);
}
.banner:before {
  content: "";
  position: absolute;
  right: -15px;
  bottom: 0;
  width: 0;
  height: 0;
  border-right: 15px solid transparent;
  border-top: 15px solid rgba(183, 183, 183, 1);
  border-bottom: 15px solid rgba(183, 183, 183, 1);
}
.banner h1 {
  align-self: center;
}
