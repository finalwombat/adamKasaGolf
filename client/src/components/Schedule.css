.schedule {
  position: absolute;
  width: 70vw;
  height: 100vh;
  background: black;
  color: white;
}

.scheduleHead {
  text-align: center;
  padding-top: 20px;
  padding-bottom: 20px;
}

.tableWrapper {
  border: none;
  border-top: 1px solid white;
  padding-top: 50px;
}
.scheduleTable {
  width: 100%;
}

thead {
  background: white;
  color: black;
  height: 20px;
}

th {
  height: 70px;
  text-align: left;
  padding: 5px;
}

tr {
  height: 70px;
}
td {
  text-align: center;
}

tr:nth-child(even) {background: black}
tr:nth-child(odd) {background: grey}
