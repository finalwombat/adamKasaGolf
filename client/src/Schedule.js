import React, { Component } from 'react'
import Background from './Background'
import backgroundImage from './img/481.jpg'

const Schedule = () => {
    return (
      <div className="schedule">
        <Background img={backgroundImage}></Background>
      </div>
    );
  }

export default Schedule
