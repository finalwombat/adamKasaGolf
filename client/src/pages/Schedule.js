import React from 'react'
import Background from '../components/Background'
import backgroundImage from '../img/481.jpg'

const Schedule = () => {
    return (
      <div className="schedule">
        <Background img={backgroundImage}></Background>
      </div>
    );
  }

export default Schedule
