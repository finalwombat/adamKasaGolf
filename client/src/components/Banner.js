import React from 'react'
import './Banner.css'

const Banner = () => {
  return (
    <div className="banner" >
      <h2>Follow Adam</h2>
    </div>
  )
}

export default Banner
