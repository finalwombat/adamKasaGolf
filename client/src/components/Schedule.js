import React from 'react'
import './Schedule.css'

export const Schedule = (props) => {
  const events = props.events.map(event => {
    return <tr>
            <td>{event.date}</td>
            <td>{event.name}</td>
            <td>{event.result}</td>
            <td><ul className="tournRounds">{scorecard(event.score)}</ul></td>
          </tr>
  })
  return (
      <div className="schedule">
        <div className="scheduleHead">
          <h1>Tournament Schedule</h1>
        </div>
        <div className="tableWrapper">
          <table className="scheduleTable">
            <thead><th>Date</th> <th>Event</th> <th>POS</th> <th>Score</th></thead>
            <tbody> { events }</tbody>
          </table>
        </div>
      </div>
  )
}


const scorecard = (score) => {
    const scores = score.split(',')
    return scores.map((score, index) => {
      console.log(index, score)
      return <li className='round'>
              <span className="roundNum">R{index + 1}</span>
              <span className="roundScore">{score}</span>
            </li>
    })
}
