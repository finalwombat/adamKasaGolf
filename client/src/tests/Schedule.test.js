import React from 'react'
import {Schedule, ScheduleEvent} from '../components/Schedule'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(<Schedule></Schedule>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

// ScheduleEvent
it('returns div with event details', () => {

  const wrapper = shallow(<ScheduleEvent date='12th sep' name='Gold Coast Open' result=''/>)
  console.log(wrapper.html())
  expect(wrapper.html()).toEqual('<div><p>12th sep</p><p>Gold Coast Open</p><p></p></div>')


})
