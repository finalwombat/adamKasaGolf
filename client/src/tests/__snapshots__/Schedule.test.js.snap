// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="scheduel"
>
  <div
    className="titles"
  >
    <h2>
      Date
    </h2>
    <h2>
      Event
    </h2>
    <h2>
      Result
    </h2>
  </div>
  <div
    className="events"
  >
    <ul>
      <li>
        <div>
          <p>
            12th sep
          </p>
          <p>
            Gold Coast Open
          </p>
          <p>
            5th
          </p>
        </div>
      </li>
      <li>
        <div>
          <p>
            16th oct
          </p>
          <p>
            Masters
          </p>
          <p>
            1st
          </p>
        </div>
      </li>
      <li>
        <div>
          <p>
            20th dec
          </p>
          <p>
            Christmass challenge
          </p>
          <p>
            25th
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
